Extension/src/LanguageServer/protocolFilter.ts

```diff
@@ -89,7 +89,17 @@ export function createProtocolFilter(): Middleware {
         provideHover: async (document, position, token, next: (document: any, position: any, token: any) => any) => clients.ActiveClient.enqueue(async () => {
             const me: Client = clients.getClientFor(document.uri);
             if (me.TrackedDocuments.has(document.uri.toString())) {
-                return next(document, position, token);
+                // Currently needed to support icons.
```

Comment:
This makes me nervous.  It looks like you're potentially hijacking any hover response, if it happens to contain an array that has a markdown string in it.

I think it would be cleaner if we switched to using a `HoverProvider` instead of hacking around the LSP hover request.  (Then you'd be more easily able to do pre/post processing of the request in TypeScript code. That may also help avoid that async/await to check if the flight is enabled, as that could happen in the a `HoverProvider`)

Reply:
Also, I think having a hover provider would allow everything to be done in the same hover request, without needing to add additional custom LSP messages.  Instead, you could add additional fields to a primary hover request message, and do any additional work immediately before or after forwarding that to the native process using `sendRequest`, like the other provider classes.

Reply:
Swapped to using the hover new hover provider pattern you added, it should be much more stable now, and we no longer need to make any changes to the intellisense hover.